modules/planning/open_space/coarse_trajectory_generator/hybrid_a_star.cc

```diff
@@ -313,6 +327,105 @@ bool HybridAStar::GenerateSpeedAcceleration(HybridAStartResult* result) {
   return true;
 }
 
+bool HybridAStar::GenerateSCurveSpeedAcceleration(HybridAStartResult* result) {
+  if (result->x.size() < 2 || result->y.size() < 2 || result->phi.size() < 2) {
+    AERROR << "result size check when generating speed and acceleration fail";
+    return false;
+  }
+
+  const size_t x_size = result->x.size();
+
+  double accumulated_s = 0.0;
+  std::vector<std::tuple<double, double, double>> s_bounds;
+  std::vector<std::tuple<double, double, double>> ds_bounds;
+
+  for (size_t i = 0; i < x_size - 1; ++i) {
+    const double discrete_v = ((result->x[i + 1] - result->x[i]) / delta_t_) *
+                                  std::cos(result->phi[i]) +
+                              ((result->y[i + 1] - result->y[i]) / delta_t_) *
+                                  std::sin(result->phi[i]);
+
+    accumulated_s += discrete_v * delta_t_;
+
+    result->v.emplace_back(discrete_v);
+    result->accumulated_s.emplace_back(accumulated_s);
+    s_bounds.emplace_back(static_cast<double>(i) * delta_t_, accumulated_s - 10,
+                          accumulated_s + 10);
+    ds_bounds.emplace_back(static_cast<double>(i) * delta_t_, discrete_v - 10,
+                           discrete_v + 10);
+  }
+
+  std::array<double, 5> w = {planner_open_space_config_.warm_start_config()
+                                 .s_curve_config()
+                                 .s_weight(),
+                             planner_open_space_config_.warm_start_config()
+                                 .s_curve_config()
+                                 .velocity_weight(),
+                             planner_open_space_config_.warm_start_config()
+                                 .s_curve_config()
+                                 .acc_weight(),
+                             planner_open_space_config_.warm_start_config()
+                                 .s_curve_config()
+                                 .jerk_weight(),
+                             planner_open_space_config_.warm_start_config()
+                                 .s_curve_config()
+                                 .ref_weight()};
+
+  std::array<double, 3> init_s = {result->accumulated_s[0], result->v[0],
+                                  (result->v[1] - result->v[0]) / delta_t_};
+
+  const size_t num_of_knots = static_cast<int>(x_size) - 1;
```

Comment:
Avoid implicit type casting.

Reply:
Sorry I missed your comment. @Capri2014 Could you fix in another pull request?